index.html

```diff
@@ -562,7 +563,75 @@ <h2>Dictionary {{RTCRtpEncodingParameters}} Members</h2>
         allows the <a>user agent</a> to adapt its bandwidth allocation strategy
         based on the current network conditions.</p>
       </dd>
+      <dt><dfn data-idl>codec</dfn> of type <span class="idlMemberType">RTCRtpCodec</span></dt>
+      <dd>
+        <p>Indicates the codec for this encoding. The {{RTCRtpCodec}} dictionary is
+          defined in [[WEBRTC]].</p>
+            <div class="issue atrisk">
+              <p>
+                {{RTCRtpEncodingParameters/codec}} is marked as a feature at risk
+                due to lack of support from implementers.
+              </p>
+            </div>
+      </dd>
     </dl>
+    <h2>Modifications to existing procedures</h2>
+    <p>Add the following steps in the parameter validation for {{RTCRtpSender/setParameters()}},
+      before any validation of <var>scalabilityMode</var> from [[WEBRTC-SVC]]:</p>
+    <ol>
+      <li>
+        <p>If <var>codecs</var> is not an empty list:</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any entry in codecs, return a promise [= rejected =] with a newly
+              [= exception/created =] {{InvalidModificationError}}.</p>
+          </li>
+        </ol>
+      <li>
+        <p>Else (<var>codecs</var> is an empty list), if the transceiver's
+          {{RTCRtpTransceiver/[[PreferredCodecs]]}} is not an empty list:</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any entry in the transceiver's {{RTCRtpTransceiver/[[PreferredCodecs]]}},
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p>
+          </li>
+        </ol>
+      </li>
+      <li>
+        <p>Else ({{RTCRtpTransceiver/[[PreferredCodecs]]}} is an empty list):</p>
+        <ol>
+          <li>
+            <p>If any <var>codec</var> parameter in <var>encodings</var> does not match
+              any codec in {{RTCRtpSender.getCapabilities(kind)}}.<code>codecs</code>,
+              return a promise [= rejected =] with a newly [= exception/created =]
+              {{InvalidModificationError}}.</p></li>
+        </ol>
+      </li>
+      <li>
+        <p>If the user agent does not support setting the codec for any encoding or mixing
+          different codec values on the different encodings, return a promise [= rejected =]
+          with a newly [= exception/created =] {{OperationError}}.
+        </p>
+      </li>
```

Comment:
In parallel, reconfigure the encoder?

Reply:
Luckily addTransceiver() will not acquire any resources, because it hasn't been negotiated yet. The spec currently says to acquire during negotiation, but as pointed out by Bernard in the last Virtual interim meeting, in practise we don't acquire encoder resources until when we actually use them, so any failure would have to be surfaced then - this is an existing problem, and perhaps we'll want to event handlers for this in the future, but that's a separate issue

Reply:
> I think the UA should be able to determine ahead of time if it is able to apply this type of configuration, so it should be able to reject it early, without a parallel step.

Depends on your definition of "apply". UAs may be able to reject some combinations early, but it's not clear to me that success is guaranteed any more strongly for customized codecs than it is for default ones.

In any case, while rejecting a promise early has some benefit while stepping through in a JS debugger, there's not much else to gain from that particular timing.